### PR TITLE
Add semantic segmentation polygon loading from Roboflow

### PR DESCRIPTION
**Pull request by capjamesg**
# Description

This PR adds support for loading the `points` values from Roboflow API detections into segmentation masks.

## Type of change

-   [X] New feature (non-breaking change which adds functionality)

## How has this change been tested, please provide a testcase or example of how you tested the change?

This code is used in `cvevals` for ingesting masks.

### CODE REVIEW

File: supervision/detection/core.py [hunk 1]
```diff
@@ -324,6 +326,28 @@ def from_roboflow(cls, roboflow_result: dict, class_list: List[str]) -> Detectio
             class_id.append(class_list.index(prediction["class"]))
             confidence.append(prediction["confidence"])
 
+            if prediction.get("points"):
+                points = prediction["points"]
+
+                mask = np.zeros((int(height), int(width)), dtype=np.uint8)
```
Comment (SkalskiP) [inline supervision/detection/core.py:332]: Most of that logic can be replaced with `polygon_to_mask` function that can create a mask given `polygon`. Please use it. You can find it in `supervision.detection.utils`.

File: supervision/detection/core.py [hunk 2]
```diff
@@ -324,6 +326,28 @@ def from_roboflow(cls, roboflow_result: dict, class_list: List[str]) -> Detectio
...
+                    [(p["x"], p["y"]) for p in points], dtype=np.int32
+                ).reshape((-1, 1, 2))
+
+                cv2.fillPoly(mask, [points], 1)
+
+                mask = mask.astype(bool)
+
+                masks.append(mask)
+
+        if len(mask) > 0:
```
Comment (SkalskiP) [inline supervision/detection/core.py:343]: Let's keep a single return and make it a bit more compact:

```python
return Detections(
    xyxy=np.array(xyxy),
    confidence=np.array(confidence),
    class_id=np.array(class_id).astype(int),
    mask=mask if mask else None,
)
```

File: supervision/detection/core.py [hunk 3]
```diff
@@ -324,6 +326,28 @@ def from_roboflow(cls, roboflow_result: dict, class_list: List[str]) -> Detectio
...
+            return Detections(
+                xyxy=np.array(xyxy),
+                confidence=np.array(confidence),
+                class_id=np.array(class_id).astype(int),
+                mask=np.array(mask),
```
Comment (SkalskiP) [inline supervision/detection/core.py:348]: Make sure that type of mask is `dtype=bool`.

File: supervision/detection/core.py [hunk 4]
```diff
@@ -324,6 +326,28 @@ def from_roboflow(cls, roboflow_result: dict, class_list: List[str]) -> Detectio
             class_id.append(class_list.index(prediction["class"]))
             confidence.append(prediction["confidence"])
 
+            if prediction.get("points"):
```
Comment (SkalskiP) [inline supervision/detection/core.py:329]: Let's prevent adding indented logic - reverse the logical condition and use `continue`.

```python
if "points" not in prediction:
    continue
```

**Comment by SkalskiP** [comment]
Adding some unit tests for `from_roboflow` would be awesome 😎

**Review by SkalskiP — CHANGES_REQUESTED** [review]

**Comment by capjamesg** [comment]
@SkalskiP I have incorporated your feedback in my last commit. Please review. I have not added a test case, however.

**Comment by SkalskiP** [comment]
@capjamesg do you have some code that I could use to test that code?

**Pull request merged**